src/multiblock_lbp.jl

```diff
@@ -0,0 +1,126 @@
+  0   0   0  0  0  0  50  50  50
+  0   0   0  0  0  0  50  50  50
+  0   0   0  0  0  0  50  50  50
+
+julia> multiblock_lbp(X, (3, 3))[5, 5] # 0b1000_0001
+0x81
+
+julia> multiblock_lbp(X, (3, 3); rotation=true)[5, 5] # 0b0000_0011
+0x03
+
+julia> multiblock_lbp(X, (3, 3); uniform_degree=2)[5, 5] # 0x09 (i.e., miscellaneous pattern)
+0x09
+```
+
+# Extended help
+
+The following is how the block-version of local binary pattern computed with
+`block_size=(3, 3)`. Check out [`lbp_original`](@ref) for more details of the original
+pixel-version of local binary pattern.
+
+```text
+2  3  2  |  7  6  6  |  2  1  3
+1  2  3  |  5  3  4  |  7  7  3
+6  5  4  |  1  4  5  |  1  5  8
+-------------------------------     sum                compare          weighted          sum
+1  2  3  |  1  7  3  |  8  3  7            28  41  37           0  1  0           0 8 0
+1  2  1  |  4  3  6  |  5  2  2     ==>    20  38  45    ==>    0  x  1     ==>   0 x 64  ==> 200
+1  2  7  |  5  3  6  |  4  7  7            33  31  39           0  0  1           0 0 128
+-------------------------------
+4  2  4  |  8  1  5  |  3  7  7
+5  7  4  |  1  1  1  |  3  4  3
+3  3  1  |  5  1  8  |  1  8  3
+```
+
+# References
+
+- [1] Zhang, Lun, et al. "Face detection based on multi-block lbp representation." _International conference on biometrics_. Springer, Berlin, Heidelberg, 2007.
+"""
+function multiblock_lbp(X::AbstractArray, block_size::Dims{2}; rotation::Bool=false, uniform_degree::Union{Nothing,Int}=nothing)
+    offsets = ((-1, -1), (0, -1), (1, -1), (-1, 0), (1, 0), (-1, 1), (0, 1), (1, 1))
+    lookups = _build_lbp_original_lookup(UInt8, 8; rotation=rotation, uniform_degree=uniform_degree)
+    multiblock_lbp!(similar(X, UInt8), X, offsets, lookups, block_size)
+end
+
+function multiblock_lbp!(out, X::AbstractMatrix, offsets, lookups, block_size::Dims{2})
+    all(isodd, block_size) || throw(ArgumentError("block size `block_size=$(block_size)` should be odd numbers."))
+    offsets = map(offsets) do o
+        o .* block_size
+    end
+    bo = CartesianIndex(block_size .÷ 2)
+
+    # It's computationally more efficient to use integral array to compute
+    # the mean block value.
+    iX = IntegralArray(X)
+
+    outerR = CartesianIndices(X)
+    r = CartesianIndex(
+        ceil(Int, maximum(abs.(extrema(first.(offsets))))),
+        ceil(Int, maximum(abs.(extrema(last.(offsets)))))
+    )
+    innerR = first(outerR)+r+bo:last(outerR)-r-bo
+
+    # TODO(johnnychen94): use LoopVectorization
+    @inbounds for I in innerR
+        # since the block is regular for inner region, doing sum is equivalent to doing average.
+        gc = iX[I-bo..I+bo]
+        # This inner loop fuses the binary pattern build stage and encoding stage for
+        # better performance.
+        rst = 0
+        # TODO(johnnychen94): it can be faster if we can enable SIMD here.
+        #                     Need to skip the boundary check in IntegralArray.
```

Comment:
Ideally, if we can make this SIMD-ed then it's likely to make `multiblock_lbp(X, (1, 1))` as fast as `lbp_origin(X)`; currently it's about 4x slower.